renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3948,6 +3948,19 @@ void ImgRefs::Split(bool splitAspects, bool splitLevels, bool splitLayers)
   areLayersSplit = newSplitLayerCount > 1;
 }
 
+QueryPoolInfo::QueryPoolInfo(WrappedVulkan *driver, VkDevice device,
+                             const VkQueryPoolCreateInfo *pCreateInfo)
+{
+  m_Buffer.Create(driver, device, pCreateInfo->queryCount * 8, 1, GPUBuffer::eGPUBufferReadback);
```

Comment:
Something I forgot before - this needs a call to `ResourceManager::SetInternalResource` to ensure this buffer is not considered a user buffer, even though it's created and wrapped. Otherwise if someone enables 'ref all resources' this buffer will show up in the capture visibly. See how these buffers are created in the text renderer for an example of how this is handled for our internal buffers during capture.